### PR TITLE
Cache chain-id

### PR DESCRIPTION
**Pull request by hweawer**
# What
According to the dashboards `chain_id` calls significantly increased in last 2 months, it is increasing usage of the provider node resources. It has huge effect because of the number of the environments we have.

# Why

Decrease usage of the provider resources

# How

Cache `chain_id`

See https://github.com/ethers-io/ethers.js/discussions/2689 for reference, a lot of providers force client to do `chainId` before fetching the block number

### CODE REVIEW

File: src/blockchain/web3_extentions/requests_metric_middleware.py [hunk 1]
```diff
@@ -50,4 +51,15 @@ def middleware(method: RPCEndpoint, params: Any) -> RPCResponse:
 
         return middleware
 
+    web3.middleware_onion.inject(
```
Comment (F4ever) [inline src/blockchain/web3_extentions/requests_metric_middleware.py:54]: add_requests_metric_middleware should not add any cache middlewares

File: src/bots/depositor.py [hunk 2]
```diff
@@ -83,16 +85,20 @@ def __init__(
                 )
             )
 
+        self._onchain_transport_w3 = None
+        self._transport_chain_id = None
         if TransportType.ONCHAIN_TRANSPORT in variables.MESSAGE_TRANSPORTS:
+            self._onchain_transport_w3 = Web3(FallbackProvider(variables.ONCHAIN_TRANSPORT_RPC_ENDPOINTS))
```
Comment (F4ever) [inline src/bots/depositor.py:91]: add cache middleware here

Comment (hweawer) [reply src/bots/depositor.py]: Done

File: src/blockchain/web3_extentions/requests_metric_middleware.py (hunk 1 above)
Comment (hweawer) [reply src/blockchain/web3_extentions/requests_metric_middleware.py:62]: Moved out of here

File: src/bots/depositor.py [hunk 3]
```diff
@@ -83,10 +82,13 @@ def __init__(
                 )
             )
 
+        self._onchain_transport_w3 = None
+        self._transport_chain_id = None
```
Comment (F4ever) [inline src/bots/depositor.py:86]: remove line

**Review by F4ever — APPROVED** [review]

File: src/bots/depositor.py (hunk 3 above)
Comment (hweawer) [reply src/bots/depositor.py]: Done

**Review by F4ever — APPROVED** [review]

**Pull request merged**